src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -0,0 +1,553 @@
+                JToken correlationGuid = run["correlationGuid"];
+
+                if (logicalId != null || description != null ||
+                    instanceGuid != null || correlationGuid != null)
+                {
+                    run.Remove("logicalId");
+                    run.Remove("description");
+                    run.Remove("instanceGuid");
+                    run.Remove("correlationGuid");
+
+                    var runId = new JObject();
+
+                    if (instanceGuid != null)
+                    {
+                        // We can only effectively populate the new instanceId in a case where
+                        // the log if previously uniquely identified a run by a guid.
+                        runId["instanceId"] = logicalId + "/" + instanceGuid;
+                    }
+
+                    if (description != null) { runId["description"] = description; }
+                    if (instanceGuid != null) { runId["instanceGuid"] = instanceGuid; }
+                    if (correlationGuid != null) { runId["correlationGuid"] = correlationGuid; }
+
+                    run["id"] = runId;
+
+                    modifiedRun = true;
+                }
+
+                // This property is an element of what v2 now refers to as aggregateIds
+                JToken automationLogicalId = run["automationLogicalId"];
+
+                if (automationLogicalId != null)
+                {
+                    run.Remove("automationLogicalId");
+
+                    var aggregateId = new JObject();
+
+                    // For the aggregating automat id, we have to provide only the
+                    aggregateId["instanceId"] = automationLogicalId + "/";
+
+                    var aggregateIds = new JArray(aggregateId);
+
+                    run["aggregateIds"] = aggregateIds;
+
+                    modifiedRun = true;
+                }
+            }
+
+            return modifiedRun;
+        }
+
+        private static bool UpdateVersionControlProvenance(JArray runs)
+        {
+            bool modifiedRun = false;
+
+            foreach (JObject run in runs)
+            {
+                JArray versionControlDetailsArray = (JArray)run["versionControlProvenance"];
+                if (versionControlDetailsArray == null) { continue; }
+
+                foreach (JObject versionControlDetails in versionControlDetailsArray)
+                {
+                    modifiedRun |= UpdateVersionControlDetails(versionControlDetails);
+                }
+            }
+            return modifiedRun;
+        }
+
+        private static bool UpdateVersionControlDetails(JObject versionControlDetails)
+        {
+            bool modifiedVersionControlDetails = false;
+
+            // https://github.com/oasis-tcs/sarif-spec/issues/242
+            modifiedVersionControlDetails |= RenameProperty(versionControlDetails, previousName: "timestamp", newName: "asOfTimeUtc");
```

Comment:
>RenameProperty [](start = 45, length = 14)

Nice! This was so smooth I almost didn't notice that you'd introduced this helper; it seemed like part of the language :-)